src/cmap/connection_pool.ts

```diff
@@ -523,6 +561,7 @@ function processWaitQueue(pool: any) {
  * @param {MongoError} error An error instance representing the error during checkout
  * @param {Connection} connection A connection from the pool
  */
+type CheckoutCallback = CallbackWithType<WaitQueueTimeoutError | MongoError, Connection>;
```

Comment:
`WaitQueueTimeoutError` should be a `MongoError`, I think we can replace this with `Callback<Connection>`